Source/loadsave.cpp

```diff
@@ -1388,7 +1389,7 @@ void SaveMonster(SaveHelper *file, Monster &monster)
 	file->WriteLE<int8_t>(monster.whoHit);
 	file->WriteLE<int8_t>(monster.level);
 	file->Skip(1); // Alignment
-	file->WriteLE<uint16_t>(monster.exp);
+	file->WriteLE<uint16_t>(monster.exp(sgGameInitInfo.nDifficulty));

```

Comment:
Should take this opportunity to fix the overflow when writing to the save file.

```suggestion
	file->WriteLE<uint16_t>(static_cast<uint16_t>(std::min<unsigned>(std::numeric_limits<uint16_t>::max(), monster.exp(sgGameInitInfo.nDifficulty))));
```